build.gradle

```diff
@@ -88,6 +103,48 @@ configurations
     integrationTestRuntime.extendsFrom testRuntime
 }
 
+protobuf {
+
+    generatedFilesBaseDir = 'gen/' 
+
+    protoc {
+        artifact = "${packages.protoc}"
+    }
+
+    generateProtoTasks {
+        all().each { task ->
+            if (task.name == "generateProto") {
+                task.outputs.upToDateWhen {
+                    new File(generatedFilesBaseDir).exists()
+                }
+            }
+        }
+
+        task clearGeneratedProto {
+            //delete generatedFilesBaseDir
+        }
+
+        task deleteGeneratedProto {
+            // the Google gradle-protobuf plugin automatically registers this task
+            // to be run as a pre-requisite to 'clean'. Since Eclipse's "Refresh Gradle Project"
```

Comment:
Potential rabbit hole: have you tried this on IntelliJ?

Reply:
No. And it's something I know needs to be done.